diff --git a/x.c b/x.c
index cd995a7..9d33361 100644
--- a/x.c
+++ b/x.c
@@ -710,6 +710,8 @@ setsel(char *str, Time t)
 	XSetSelectionOwner(xw.dpy, XA_PRIMARY, xw.win, t);
 	if (XGetSelectionOwner(xw.dpy, XA_PRIMARY) != xw.win)
 		selclear();
+
+	clipcopy(NULL);
 }
 
 void
@@ -726,6 +728,10 @@ brelease(XEvent *e)
 		return;
 	}
 
+	if (e->xbutton.button == Button2)
+		clippaste(NULL);
+	else if (e->xbutton.button == Button1)
+		mousesel(e, 1);
 	if (mouseaction(e, 1))
 		return;
 	if (e->xbutton.button == Button1)
