diff --git a/config.def.h b/config.def.h
index d2fdfc2..acb90bf 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "Liberation Mono:pixelsize=24:antialias=true:autohint=true";
 static int borderpx = 2;

 /*
@@ -34,7 +34,7 @@ static float chscale = 1.0;
  *
  * More advanced example: L" `'\"()[]{}"
  */
-wchar_t *worddelimiters = L" ";
+wchar_t *worddelimiters = L" '`\"()[]{}<>|";

 /* selection timeouts (in milliseconds) */
 static unsigned int doubleclicktimeout = 300;
@@ -133,7 +133,7 @@ static const char *colorname[] = {
  */
 unsigned int defaultfg = 7;
 unsigned int defaultbg = 258;
-static unsigned int defaultcs = 256;
+static unsigned int defaultcs = 1;
 static unsigned int defaultrcs = 257;

 /*
@@ -156,7 +156,7 @@ static unsigned int rows = 24;
  * Default shape of the mouse cursor
  */

-static char* mouseshape = "xterm";
+static char* mouseshape = "xterm";

 /*
  * Color used to display font attributes when fontconfig selected a font which
diff --git a/config.mk b/config.mk
index 43d3b49..1f25c54 100644
--- a/config.mk
+++ b/config.mk
@@ -23,12 +23,12 @@ LIBS = -L$(X11LIB) -lm -lrt -lX11 -lutil -lXft -lXrender -lXcursor \
        `$(PKG_CONFIG) --libs harfbuzz`

 # flags
-STCPPFLAGS = -DVERSION=\"$(VERSION)\" -D_XOPEN_SOURCE=600
-STCFLAGS = $(INCS) $(STCPPFLAGS) $(CPPFLAGS) $(CFLAGS)
-STLDFLAGS = $(LIBS) $(LDFLAGS)
+STCPPFLAGS += -DVERSION=\"$(VERSION)\" -D_XOPEN_SOURCE=600
+STCFLAGS += $(INCS) $(STCPPFLAGS) $(CPPFLAGS) $(CFLAGS)
+STLDFLAGS += $(LIBS) $(LDFLAGS)

 # OpenBSD:
-#CPPFLAGS = -DVERSION=\"$(VERSION)\" -D_XOPEN_SOURCE=600 -D_BSD_SOURCE
+#CPPFLAGS += -DVERSION=\"$(VERSION)\" -D_XOPEN_SOURCE=600 -D_DEFAULT_SOURCE
 #LIBS = -L$(X11LIB) -lm -lX11 -lutil -lXft \
 #       `$(PKG_CONFIG) --libs fontconfig` \
 #       `$(PKG_CONFIG) --libs freetype2`
